File: src/3-Problem03.cpp
#include <iostream>
using namespace std;

bool is_composite(int n);

int main () {
    int A, B, min, max;
    cout << "Masukkan nilai A : ";
    cin >> A;
    cout << "Masukkan nilai B : ";
    cin >> B;
    
    if (A > B) {
        max = A;
        min = B;
    } else {
        min = A;
        max = B;
    }
    cout << "Bilangan komposit diantara " << min << " sampai " << max << " adalah :" << endl;
    for (int i = min+1; i < max; i++) {
        if (is_composite(i)) {
            cout << i << endl;
        }
    }
    return 0;
}

bool is_composite(int n) {
    if (n <= 1) {
        return false;
    }
    else if (n <= 3) {
        return false;
    } else if ((n % 2 == 0)||(n % 3 == 0)) {
        return true;
    }
    int i = 5;
    while (i*i <= n) {
        if ((n % i == 0)||(n % (i+2) == 0)) {
            return true;
        }
        i += 6;
    }
    return false;
}

File: src/3-Problem03.py
import math

def is_composite(x):
    result = False
    factors_count = 0

    if x > 2 :
        for i in range(2, int(math.floor(math.sqrt(x))) + 1) :
            if x % i == 0 : factors_count += 1
        result = factors_count >= 1

    return result

if __name__ == "__main__":
    nilai = [0 for _ in range(2)]
    nilai[0] = int(input("Masukkan nilai A :"))
    nilai[1] = int(input("Masukkan nilai B :"))

    print("Bilangan komposit di antara {} sampai {} adalah:".format(min(nilai), max(nilai)))

    for x in range(min(nilai), max(nilai) + 1) :
        if is_composite(x) : print(x)

